### ISSUE TITLE
E-mail embedded images

### ISSUE THREAD

**Issue by jansarmir** [pimcore/pimcore#407]
These days e-mails should look nice, but remote images make them ugly ;]

It would be nice to have option for embedding images inline into into email or it could be standard.
[Mod could be done easy](http://www.developerfiles.com/inserting-images-in-email-with-zend-framework/).

The mail system has some other implications like: logging, view sent email, resend, ...
So i analysed it and
* embedded option have to be stored (resending email should do the same)
* logging should be done before sending (we dont want to log html with CIDs)
** btw, why is this after? for performance, problems?
* next embedded image modification should be processed
* and finally send email
I think it should be all, or did i miss something?

**Comment by jansarmir** [comment]
It is slower, BUT e-mail is perfect ;]
I've moded mod above to solve some issues
```php
    public function embeddedHtml($bodyHtmlRendered)
    {
        $this->setType(\Zend_Mime::MULTIPART_RELATED);
    
        $matches = array();
        preg_match_all('~<img[^>]+src\s*=\s*(\'|")(\w++://.*?)\\1~is', $bodyHtmlRendered, $matches);
    
        foreach (array_unique($matches[2]) as $filename) {
            $data = @file_get_contents($filename);
            if($data === false) {
                continue;
            }
            $at = $this->createAttachment($data);
            $at->type = $this->mimeByExtension($filename);
            $at->disposition = \Zend_Mime::DISPOSITION_INLINE;
            $at->encoding = \Zend_Mime::ENCODING_BASE64;
            $at->id = 'cid_' . md5_file($filename);
            $str = preg_quote($filename);
            $bodyHtmlRendered = preg_replace("~('|\")$str('|\")~", 'cid:' . $at->id, $bodyHtmlRendered);
        }
        return $bodyHtmlRendered;
    }
```

**Comment by brusch** [comment]
I don't think this is a good idea, see also: https://sendgrid.com/blog/embedding-images-emails-facts/ 
This causes especially problem with web-based email clients.

**Comment by bbak** [comment]
Even tough I agree to sendgrid's view in general, IMHO the System shouldn't impose a restriction and leave that decision up to the User or Client - Clients sometime insist to have Images embedded (for reasons beyond the scope of this comment).
Even the Sendgrid article states, that there's no _unified support_ for any of the three options mentioned.
